compiler/transforms/memref_to_snax.py

```diff
@@ -0,0 +1,126 @@
+from xdsl.dialects import builtin, memref
+from xdsl.dialects.arith import Addi, Constant, Muli
+from xdsl.dialects.builtin import IndexType, NoneAttr, UnrealizedConversionCastOp
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+from compiler.dialects import snax
+from compiler.dialects.tsl import TiledStridedLayoutAttr
+
+
+class AllocOpRewrite(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, alloc_op: memref.Alloc, rewriter: PatternRewriter):
+        """Swap memref.alloc op with snax.alloc, for now, we suppport
+        NoneType layouts and TSL Layouts"""
+
+        # a (very) temporary constraint: only allow for static sizes
+        # i am already working on the dynamic case
+        if any([x.data == -1 for x in alloc_op.memref.type.shape.data]):
+            return
+
+        # get the memref type
+        memref_type: memref.MemRefType = alloc_op.memref.type
+
+        # get the element type
+        element_type = memref_type.get_element_type()
+
+        if not isinstance(element_type, builtin.IntegerType):
+            return
```

Comment:
Why is this a requirement? f64 and i64 have the same size, and thus allocation requirement, no?